com.unity.ml-agents.extensions/Runtime/Sensors/GridSensor.cs

```diff
@@ -669,83 +734,100 @@ protected virtual void ValidateValues(float[] channelValues, GameObject currentC
         /// </summary>
         /// <param name="currentColliderGo">The game object that was found colliding with a certain cell</param>
         /// <param name="cellIndex">The index of the current cell</param>
-        /// <param name="normalized_distance">A float between 0 and 1 describing the ratio of
+        /// <param name="normalizedDistance">A float between 0 and 1 describing the ratio of
         ///            the distance currentColliderGo is compared to the edge of the gridsensor</param>
-        protected virtual void LoadObjectData(GameObject currentColliderGo, int cellIndex, float normalized_distance)
+        protected virtual void LoadObjectData(GameObject currentColliderGo, int cellIndex, float normalizedDistance)
         {
-            for (int i = 0; i < DetectableObjects.Length; i++)
+            Profiler.BeginSample("GridSensor.LoadObjectData");
+            unsafe
             {
-                if (currentColliderGo != null && currentColliderGo.CompareTag(DetectableObjects[i]))
+                var channelHotVals = stackalloc float[ObservationPerCell];
```

Comment:
Here's where I am allocating a stack buffer for use, where previously a GC allocation was made within the loop.

Reply:
only skimmed the code, but instead of doing this or the old approach, can we have the GridDepthType.ChannelHot code write to m_PerceptionBuffer directly? Or better, make an ArraySegment for the part of the m_PerceptionBuffer that this cellIndex is allowed to write to? This would avoid the stack allocation and memcpy calls (in the GridDepthType.Channel too).